File: Math/QuadraticFormula.py
# Required for math.sqrt
from math import sqrt
# Required to end program if error occurs
import sys

# Takes user input and gets numbers
a = float(input('Please enter a number to represent a: '))
if a == 0:
    sys.exit('ERROR: a cannot be equal to zero.')
b = float(input('Please enter a number to represent b: '))
c = float(input('Please enter a number to represent c: '))

# Formula to calculate roots
x1 = (-b + sqrt((b**2) - 4*a*c))/(2*a)
x2 = (-b - sqrt((b**2) - 4*a*c))/(2*a)

# Prints out values
print('The Quadratic Roots are: ', x1,  ' and ', x2)

File: Math/CPP/QuadraticFormula.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main() {
 	for( ; ; ) {
  		double a,b,c;
 	
  		// Gets coefficients
  		cout <<"Enter the coefficients starting with a then b then c \n";
		cin>>a>>b>>c;
  
 	 	if (a==0.0){
     		cout <<"Undefined \n";
  			return 1;
      }
      	// Quadratic Formula Finder
  		double x1, x2;
    	x1 =(-b + sqrt((b*b) - 4*a*c))/(2*a);
    	x2 =(-b - sqrt((b*b) - 4*a*c))/(2*a); 
   		cout <<"The roots are <>x1</b> = " << x1 << " and x2 = " << x2 << "\n \n";
   }
}
